allennlp/modules/seq2seq_decoders/seq_decoder.py

```diff
@@ -0,0 +1,38 @@
+from typing import Dict
+
+import torch
+from torch.nn import Module
+
+from allennlp.common import Registrable
+from allennlp.modules import Embedding
+
+
+class SeqDecoder(Module, Registrable):
+    """
+    A ``SeqDecoder`` is a base class for different types of Seq decoding modules
```

Comment:
Could you elaborate on this a bit? I think it's important that users reading this and `DecoderNet` be able to immediately understand the purpose of each. You could even some something like, "In contrast SeqDecoder, DecoderNet is intended for..."